libkbfs/journal_server.go

```diff
@@ -482,12 +484,37 @@ func (j *JournalServer) enableLocked(
 	return tj, nil
 }
 
-// Enable turns on the write journal for the given TLF.
+// Enable turns on the write journal for the given TLF.  If h is nil,
+// it will be attempted to be fetched from the remote MD server.
 func (j *JournalServer) Enable(ctx context.Context, tlfID tlf.ID,
-	bws TLFJournalBackgroundWorkStatus) error {
+	h *TlfHandle, bws TLFJournalBackgroundWorkStatus) (err error) {
 	j.lock.Lock()
 	defer j.lock.Unlock()
-	tj, err := j.enableLocked(ctx, tlfID, bws, false)
+	var tid keybase1.TeamID
+	if tlfID.Type() == tlf.SingleTeam {
+		if h == nil {
+			// We must have a handle for SingleTeam TLFs, so try to
+			// get one from the server.  This is an unstrusted mapping
+			// since it's just using whatever the server tells it.
```

Comment:
s/unstrusted/untrusted

Also, would it be worthwhile to store whether the handle is trusted somewhere in the handle?